site/source/docs/porting/exceptions.rst

```diff
@@ -32,19 +23,53 @@ For example, build the following program with ``emcc -fexceptions``:
       return 0;
     }
 
-Running it will print ``throw...`` and then ``catch!``.
+The first ``throw`` will abort the program and you'll see something like this in the output:
+
+.. code-block:: text
+
+  throw...  
+  exception thrown: 5246024 - Exception catching is disabled, this exception cannot be caught. Compile with -s NO_DISABLE_EXCEPTION_CATCHING or -s EXCEPTION_CATCHING_ALLOWED=[..] to catch.
+
+If you want to opt-in, you have two following options.
+
+JavaScript-based exception support
+##################################
+
+First, you can enable exceptions via Emscripten's JavaScript-based support.
+
+To enable it, pass ``-fexceptions`` at both compile time and link time.
+
+When you rebuild the example above with this flag, the output will change to:
+
+.. code-block:: text
+
+  throw...
+  catch!
+
+Note that this option has relatively high overhead, but it will work on all JavaScript
+engines with WebAssembly support. You can reduce the overhead by specifying a
+list of allowed functions in which exceptions are enabled, see the
+``EXCEPTION_CATCHING_ALLOWED`` setting.
+
+WebAssembly exception handling proposal
+#######################################
+
+Alternatively, you can opt-in to the `native WebAssembly exception handling
```

Comment:
maybe remove "the" or add "feature" or perhaps "extension" at the end of the sentence?

Reply:
Added "extension".

Reply:
Have we used the word 'extension' to describe a new proposal before? It can be little confusing.. How about 'feature' or just 'proposal'?

Reply:
Heh, avoiding having to choose the right word here is exactly why I omitted any word in the first place 😀

Reply:
I picked "extension" because it seemed likely to make more sense to someone who doesn't know anything about the wasm standards process.

Reply:
I changed to "proposal", but happy to revert if that doesn't work.